bin/knc.c

```diff
@@ -869,6 +869,7 @@ write_local_buffer(work_t *work)
 			 */
 			LOG(LOG_DEBUG, ("write got EPIPE"));
 
+			work->local_buffer.in_valid = 0;
```

Comment:
Per your explanation, this fixes something of a livelock/timeout because we don't `shutdown()` if there's pending data.  Not sure this is the best way to fix that, but I'll take it, but I do want the commit comment to explain the observable behavior -- that this commit fixes a possible timeout.